source/Engine.cpp

```diff
@@ -1582,8 +1646,9 @@ void Engine::CalculateStep()
 		DoWeather(weather);
 
 	// Check for flotsam collection (collisions with ships).
-	for(const shared_ptr<Flotsam> &it : flotsam)
+	for_each_mt(flotsam.begin(), flotsam.end(), [&](const auto &it) {
 		DoCollection(*it);
+	});
```

Comment:
and again

Reply:
I may not remember how useful every single change is, but this one does actually improve performance when you have flotsam flying around.